File: cloud-functions/submitTrade/exchanges/bingx/scripts/cancel.js
const CustomError = require('../../../utils/error.js');
const { getSpecificOrder, deleteTpSlOrder } = require('../../../utils/firestore.js');


async function sendCancel(session, traderId, tradeId, documentId, tradeType) {
    try {
        let order = await getSpecificOrder(traderId, tradeId, documentId, tradeType)

        await session.cancelOrder(Symbol, null, order.orderId)
        await deleteTpSlOrder(traderId, tradeId, documentId, tradeType)

    } catch(error) {
        throw new CustomError({
            message: `Error cancelling order in scripts: ${error.message}`,
            status: 500,
            source: 'sendCancel',
        });
    }
}

module.exports = sendCancel;

File: cloud-run/exec-handlers/SUB_BINGX/bingx_api/scripts/cancel.py
from utils.firestore import get_specific_order, delete_tp_sl_order

async def send_cancel(session, symbol, user_id, trader_id, trade_id, document_id, trade_type):
    order = await get_specific_order(trader_id, user_id, trade_id, document_id, trade_type)

    await session.cancel_order(symbol, order['orderID'], None)
    await delete_tp_sl_order(trader_id, user_id, trade_id, document_id, trade_type)

    return